src/python/pants/backend/python/lint/pylint/rules_integration_test.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pathlib import PurePath
+from textwrap import dedent
+from typing import List, Optional
+
+import pytest
+
+from pants.backend.python.lint.pylint.rules import PylintTarget
+from pants.backend.python.lint.pylint.rules import rules as pylint_rules
+from pants.backend.python.rules import (
+  download_pex_bin,
+  inject_init,
+  pex,
+  prepare_chrooted_python_sources,
+)
+from pants.backend.python.subsystems import python_native_code, subprocess_environment
+from pants.backend.python.targets.python_library import PythonLibrary
+from pants.build_graph.address import Address
+from pants.build_graph.build_file_aliases import BuildFileAliases
+from pants.engine.fs import FileContent, InputFilesContent, Snapshot
+from pants.engine.legacy.structs import PythonTargetAdaptor
+from pants.engine.rules import RootRule
+from pants.engine.selectors import Params
+from pants.rules.core import strip_source_root
+from pants.rules.core.lint import LintResult
+from pants.source.wrapped_globs import EagerFilesetWithSpec
+from pants.testutil.option.util import create_options_bootstrapper
+from pants.testutil.test_base import TestBase
+
+
+class PylintIntegrationTest(TestBase):
+  source_root = "test"
```

Comment:
I think you'll want this to be longer like `src/python/pants`. We want to ensure that we properly strip the source root.